src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -136,7 +137,7 @@ def invoke_package_manager(self, command):
 
                 if i < self.package_manager_max_retries - 1:
                     self.composite_logger.log_warning("Exception on package manager invoke. [Exception={0}] [RetryCount={1}]".format(error_msg, str(i)))
-                    time.sleep(i + 1)
+                    time.sleep(pow(2, i + 2))
```

Comment:
Function invoke_package_manager is handling a lot of things and has become too long. Think on extracting out soem part, eg: the retries could be in a different function called from invoke_package_manager

Reply:
Good idea. I split it up into a few separate functions.